### PR TITLE
Homework7: Andrei Kulbitski

### PR DESCRIPTION
**Pull request by kulbitski-andrei**
Three tasks added:
1. Bulls and Cows;
2. Pyramid;
3. Statuesques.

### CODE REVIEW

File: homework_7.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    print("*" * starfield)
+
+    var_N -= 1
+    starfield += 2
+
+# 3. Статуи¶
+print("\nTASK 3: STATUESQUES")
+
+
+def birthday_presents_sorted_func(*args):
```
Comment (Stml89) [inline homework_7.py:93]: in general you do not need to sort the list, just use min and max values from the given list

File: homework_7.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+    if count_of_bulls == 4:
+        # Victory condition is met
+        print("Congratulations, you have won!")
+        break
+
+    attempts -= 1
+
+    if attempts <= 0:
+        # Victory condition is not met
```
Comment (Stml89) [inline homework_7.py:67]: pls do not write comments for obvious actions

File: homework_7.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    digits = [1, 2, 3, 4, 5, 6, 7, 8, 9, 0]
+    random.shuffle(digits)
+    return digits[:4]
+
+
+# Magic number is stored as four integers packaged in the list
+magic_number = magic_number_generator_func()
+
+# User is expected to guess the number in four attempts :)
+attempts = 8
```
Comment (Stml89) [inline homework_7.py:20]: can not this information in the requirements

File: homework_7.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    position_key = 0
+    # Bulls - amount of full alignments (digit and position)
+    count_of_bulls = 0
+    # Cows - amount of partial alignments (digit only)
+    count_of_cows = 0
+
+    # Handling user input:
+    # Every non-4-digit input will be rejected
+    # and user would be forced to retry
+    current_attempt = input(
```
Comment (Stml89) [inline homework_7.py:33]: pls add verification that user enter unique number

File: homework_7.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        current_attempt_list.append(int(i))
+
+    for digit_b in magic_number:
+        # Loop for calculation of amount of bulls
+        if digit_b == current_attempt_list[position_key]:
+            count_of_bulls += 1
+        position_key += 1
+    print("Bulls:", count_of_bulls)
+
+    for digit_c in magic_number:
```
Comment (Stml89) [inline homework_7.py:53]: cows calculation does not work correctly

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework_7.py (hunk 3 above)
Comment (kulbitski-andrei) [reply homework_7.py:20]: attempts removed

File: homework_7.py (hunk 1 above)
Comment (kulbitski-andrei) [reply homework_7.py]: Ok, removed sorting function

File: homework_7.py (hunk 5 above)
Comment (kulbitski-andrei) [reply homework_7.py:53]: Fixed it, now: count_of_cows = cows - bulls

**Review by Stml89 — APPROVED** [review]

**Pull request merged**